llmfoundry/data/text_data.py

```diff
@@ -240,6 +240,9 @@ def build_text_dataloader(
             # fwd all kwargs with **stream allows streaming to check args
             streams.append(Stream(**stream))
 
+    import streaming.base.util
+    streaming.base.util.clean_stale_shared_memory()
```

Comment:
Is this safe?

What happens if I'm running 2 jobs on the same instance / gpu? Will the second one I launch kill the first since it'll clean up its dataloader?

Reply:
Actually, yeah I agree with vitaliy. Can you do this cleanup in the test only please?

Reply:
@karan6181 can you chime in on if this command is safe for 2 jobs? 

For more context, I couldn't create both the streaming dataset and run the training pipeline without errors without this line (most likely because stale shared memory was left over).

Reply:
If we are running more than one job, calling `streaming.base.util.clean_stale_shared_memory()` would be devastating, it will close existing shared memory which is an unintended behavior. Ideally, one should call this method only once during their entire run. 

>For more context, I couldn't create both the streaming dataset and run the training pipeline without errors without this line (most likely because stale shared memory was left over).

If the job finishes successfully, then I would expect perfect clean up and you can run the same job on the same machine again. What error do you see when you don't include `streaming.base.util.clean_stale_shared_memory()` ?

Reply:
Thanks @karan6181 resolved offline by creating separate datasets on local disk based on the test device